magefile.go

```diff
@@ -49,3 +50,43 @@ func (Lint) Codegen(ctx context.Context) error {
 		return nil
 	})
 }
+
+type Build mg.Namespace
+
+// Dagger will build the dagger binary
+func (Build) Dagger(ctx context.Context) error {
+	return engine.Start(ctx, nil, func(ctx engine.Context) error {
+		core := api.New(ctx.Client)
+
+		builder := core.Container().From("golang:1.18.2-alpine")
+
+		src, err := core.Host().Workdir().Read().ID(ctx)
+		if err != nil {
+			return err
+		}
+
+		builder = builder.WithMountedDirectory("/app", src).WithWorkdir("/app")
+
+		builder = builder.Exec(api.ContainerExecOpts{
+			Args: []string{"mkdir", "/app/build"},
```

Comment:
@vito is there a way to do a mkdir through the API?

Reply:
Maybe this could be merged with #3264 -- `withNewDirectory()` without providing a `DirectoryID` creates an empty dir?

Reply:
I'd rather merge this now, with a note on #3264 to update it with the new API.
It will give a real use case to try on.

Reply:
Oops, missed this question, yeah we'll need #3264 for that. I like either adding `withNewDirectory` or just making the `DirectoryID` optional in #3264.